src/util/primitives.js

```diff
@@ -204,6 +218,26 @@ class Aabb {
         return new Aabb(qMin, qMax);
     }
 
+    applyTransform(transform: Mat4): Aabb {
+        const corners = this.getCorners();
+
+        for (let i = 0; i < corners.length; ++i) {
+            vec3.transformMat4(corners[i], corners[i], transform);
+        }
+
+        this.min = [Infinity, Infinity, Infinity];
+        this.max = [-Infinity, -Infinity, -Infinity];
+
+        for (const p of corners) {
+            vec3.min(this.min, this.min, p);
+            vec3.max(this.max, this.max, p);
+        }
+
+        this.center = vec3.scale([], vec3.add([], this.min, this.max), 0.5);
+
+        return this;
```

Comment:
If you make this method immutable, you could remove some repetitive code:

```suggestion
        return Aabb.fromPoints(corners);
```